CMakeLists.txt

```diff
@@ -61,16 +61,18 @@ set(GEN_TABLES
 
 if(MSVC)
 else()
-    # Do not use -march=native for distributed compilation.
-    # Run g++ -Q -march=native --help=target and get the native architecture.
-    execute_process(COMMAND g++ -Q -march=native --help=target
-                    COMMAND grep -- -march
-                    COMMAND head -n1
-                    COMMAND sed "s/-march=//g"
-                    COMMAND sed "s/[ \t]*//g"
-                    OUTPUT_VARIABLE DETAILED_ARCHITECTURE
-                    OUTPUT_STRIP_TRAILING_WHITESPACE)
-    set(CMAKE_CXX_FLAGS "-Wall -Wextra -march=${DETAILED_ARCHITECTURE}")
+  if(CMAKE_CXX_COMPILER_VERSION VERSION_GREATER_EQUAL 9)
+        # Do not use -march=native for distributed compilation.
+        # Run g++ -Q -march=native --help=target and get the native architecture.
+        execute_process(COMMAND g++ -Q -march=native --help=target
+                       COMMAND grep -- -march
+                       COMMAND head -n1
+                       COMMAND sed "s/-march=//g"
+                       COMMAND sed "s/[ \t]*//g"
+                       OUTPUT_VARIABLE DETAILED_ARCHITECTURE
+                       OUTPUT_STRIP_TRAILING_WHITESPACE)
+        set(CMAKE_CXX_FLAGS "-Wall -Wextra -march=${DETAILED_ARCHITECTURE}")
```

Comment:
I think we should keep `-Wall -Wextra` even if we use gcc-8.

Reply:
Addressed.